samples/extensions/full_screen_exclusive/full_screen_exclusive.cpp

```diff
@@ -0,0 +1,1201 @@
+						application_window_status                                  = ApplicationWindowMode::Windowed;
+						surface_full_screen_exclusive_info_EXT.fullScreenExclusive = VK_FULL_SCREEN_EXCLUSIVE_DISALLOWED_EXT;
+						isRecreate                                                 = true;
+						isFullScreenExclusive                                      = false;
+						LOGI("Windowed Mode Detected!")
+					}
+					break;
+				case vkb::KeyCode::F2:        // FullscreenExclusiveEXT = Allowed
+					if (full_screen_status != SwapchainMode::BorderlessFullscreen)
+					{
+						full_screen_status                                         = SwapchainMode::BorderlessFullscreen;
+						application_window_status                                  = ApplicationWindowMode::Fullscreen;
+						surface_full_screen_exclusive_info_EXT.fullScreenExclusive = VK_FULL_SCREEN_EXCLUSIVE_ALLOWED_EXT;
+						isRecreate                                                 = true;
+						isFullScreenExclusive                                      = false;
+						LOGI("Borderless Fullscreen Mode Detected!")
+					}
+					break;
+				case vkb::KeyCode::F3:        // FullscreenExclusiveEXT = Application Controlled
+					if (full_screen_status != SwapchainMode::ExclusiveFullscreen)
+					{
+						full_screen_status                                         = SwapchainMode::ExclusiveFullscreen;
+						application_window_status                                  = ApplicationWindowMode::Fullscreen;
+						surface_full_screen_exclusive_info_EXT.fullScreenExclusive = VK_FULL_SCREEN_EXCLUSIVE_APPLICATION_CONTROLLED_EXT;
+						isRecreate                                                 = true;
+						isFullScreenExclusive                                      = true;
+						LOGI("Exclusive Fullscreen Mode Detected!")
+					}
+					break;
+				default:        // FullscreenExclusiveEXT = Default
+					isRecreate = false;
+					break;
+			}
+
+			// now if to recreate the swapchain and everything related:
+			if (isRecreate)
+			{
+				FullScreenExclusive::recreate();
+			}
+		}
+	}
+#endif
+}
+
+void FullScreenExclusive::update_application_window()
+{
+#if defined(VK_USE_PLATFORM_WIN32_KHR)
+	if (application_window_status == ApplicationWindowMode::Fullscreen && isWindowed)        // check if it is already in fullscreen, if is, then do nothing
+	{
+		isWindowed = false;
+
+		GetWindowPlacement(HWND_applicationWindow, &wpc);
+		if (HWND_style == 0)
+			HWND_style = GetWindowLong(HWND_applicationWindow, GWL_STYLE);
+		if (HWND_style_previous == 0)
+			HWND_style_previous = GetWindowLong(HWND_applicationWindow, GWL_EXSTYLE);
+
+		long HWND_newStyle = HWND_style;
+		HWND_newStyle &= ~WS_BORDER;
+		HWND_newStyle &= ~WS_DLGFRAME;
+		HWND_newStyle &= ~WS_THICKFRAME;
+
+		long HWND_newStyle_previous = HWND_style_previous;
+		HWND_newStyle_previous &= ~WS_EX_WINDOWEDGE;
+
+		SetWindowLong(HWND_applicationWindow, GWL_STYLE, HWND_newStyle | static_cast<long>(WS_POPUP));
```

Comment:
Why do you prepare the styles above, and add some bits here? Should probably be done in one block, for each of the styles.